lumicks/pylake/tests/test_imaging_camera/test_correlated_stack.py

```diff
@@ -399,6 +399,44 @@ def test_correlated_stack_plotting(rgb_alignment_image_data):
     with pytest.raises(IndexError, match="Frame index out of range"):
         stack.plot(channel="blue", frame=-1)
 
+    # Test recognition of and plotting with deprecated order of arguments
+    # old arguments: frame, channel, show_title, axes, adjustment
+    # old_arg_types = [int, str, bool, matplotlib.axes.Axes, ColorAdjustment]
+    # nones_allowed = [False, True, False, True, False]
+    for args in [
+        [0],
+        [1, None],
+        [0, "blue"],
+        [0, None, True],
+        [0, None, True, "SUPPLY_AXES"],  # "SUPPLY_AXES" will be replaced with an mpl.axes object
+        [0, None, True, None, ColorAdjustment.nothing()]
+    ]:
+        frame = args[0]
+        channel = "blue" if args[-1] == "blue" else "rgb"
+        args[-1] = plt.subplots()[1] if args[-1] == "SUPPLY_AXES" else args[-1]
```

Comment:
Why not just put the axes object in immediately?

Reply:
This is due to the internal state of matplotlib keeping the current axes (usually the one created recently). Calling `plt.close()` will per default close the figure belonging to the current axes. As `plt.close()` is called in the for loop several times before "SUPPLY_AXES" is evaluted, the axes created before the for loop would have been closed by then.
I added a comment explaining why this is done as it is.